### ISSUE TITLE
PX-2: Create Phoenix Evaluation Dataset

### ISSUE THREAD

**Issue by EvieHwang** [EvieHwang/wm2#82]
## Phoenix Implementation - Phase 2: Test Data

**Reference:** [WM2 APF Evaluation Criteria](https://github.com/EvieHwang/wm2/blob/main/docs/wm2-apf-criteria.md)
**Dependencies:** #81 (PX-1: Setup)

---

### Purpose

Create a Phoenix Dataset containing labeled test cases for WM2 classification evaluation. Phoenix datasets can be created from CSV, Pandas DataFrames, or extracted from traces.

#### 1. Dataset Design
- [ ] Decide holdout size (recommendation: 50-100 cases)
- [ ] Stratify across all 5 categories (Pouch, Small Bin, Tote, Carton, Oversized)
- [ ] Include edge cases with documented acceptable alternatives
- [ ] Ensure no overlap with ChromaDB training data

#### 2. Dataset Format

Phoenix datasets use a simple structure. Each row has:
- `input`: The data passed to your task
- `expected` (optional): The expected/reference output
- `metadata` (optional): Additional context

```python
# Example dataset format for WM2
dataset_rows = [
    {
        "input": {"description": "iPhone 15 Pro Max 256GB"},
        "expected": {"classification": "TOTE"},
        "metadata": {"category": "electronics", "is_edge_case": False}
    },
    {
        "input": {"description": "Small USB-C cable 6 inches"},
        "expected": {"classification": "POUCH"},
        "metadata": {
            "category": "accessories", 
            "is_edge_case": True, 
            "acceptable_alternatives": ["SMALL_BIN"]
        }
    }
]
```

#### 3. Create Dataset in Phoenix

**Option A: From CSV**
```python
# Upload via Phoenix UI
# CSV columns: description, expected_classification, category, is_edge_case
```

**Option B: From Pandas (Recommended)**
```python
import pandas as pd
from phoenix.client import Client

client = Client()

# Load from existing test data
df = pd.read_csv("backend/data/holdout_test_set.csv")

# Transform to Phoenix format
dataset_df = pd.DataFrame({
    "input": df["description"].apply(lambda x: {"description": x}),
    "expected": df["classification"].apply(lambda x: {"classification": x}),
    "metadata": df.apply(lambda row: {
        "category": row.get("category", "unknown"),
        "is_edge_case": row.get("is_edge_case", False)
    }, axis=1)
})

# Upload to Phoenix
dataset = client.datasets.upload(
    name="wm2-holdout",
    dataframe=dataset_df,
)
print(f"Created dataset: {dataset.id}")
```

**Option C: From Traces (Production Data)**
```python
# Export spans from a Phoenix project
spans_df = client.spans.get_spans_dataframe(
    project_identifier="wm2-production"
)
# Filter and transform into dataset
```

#### 4. Dataset Versioning

Phoenix supports dataset versioning. Tag important versions:
```python
# After uploading
client.datasets.update(
    dataset_id=dataset.id,
    description="v1.0 - Initial holdout set, 100 cases, stratified"
)
```

### Key Decisions

- **Stratification:** Equal distribution across categories (not proportional to real-world)
- **Edge cases:** ~20% of dataset should be intentionally ambiguous
- **Ground truth source:** Human-validated labels (not raw CSV)

### Acceptance Criteria

- [ ] Dataset created in Phoenix
- [ ] 50+ labeled test cases
- [ ] All 5 categories represented
- [ ] Edge cases identified and documented
- [ ] Dataset accessible via Phoenix client
- [ ] Ready to proceed to PX-3

### Resources

- [Phoenix Docs: Creating Datasets](https://docs.arize.com/phoenix/datasets-and-experiments/how-to-datasets/creating-datasets)
- [Phoenix Docs: Overview Datasets](https://docs.arize.com/phoenix/datasets-and-experiments/overview-datasets)

**Comment by EvieHwang** [comment]
## Implementation Complete

Branch: `006-arize-eval-dataset`
PR: Ready for creation at https://github.com/EvieHwang/wm2/pull/new/006-arize-eval-dataset

### Deliverables

**Script**: `eval/scripts/generate_eval_dataset.py`
- Parses product dimensions and weights from reference CSV
- Classifies products using dimension/weight thresholds
- Generates 60-case evaluation dataset with target distribution
- CLI with `--input`, `--output`, `--seed`, `--verbose` options
- Deterministic output via seeded random sampling

**Dataset**: `eval/datasets/wm2_eval_v1.csv`
| Category | Count | Percentage |
|----------|-------|------------|
| Pouch | 24 | 40% |
| Small Bin | 15 | 25% |
| Tote | 12 | 20% |
| Carton | 6 | 10% |
| Oversized | 3 | 5% |
| **Total** | **60** | 100% |

Edge cases: 57 (95%) - prioritized products near category boundaries

**Tests**: 32 unit tests covering:
- Dimension parsing (6 tests)
- Weight parsing (6 tests)
- Container classification (8 tests)
- Data loading and sampling (4 tests)
- Edge case prioritization (4 tests)
- CLI and reproducibility (4 tests)

### Usage
```bash
python eval/scripts/generate_eval_dataset.py --verbose
```

### Arize Integration
Dataset includes columns for Phoenix schema:
- `example_id`, `input`, `ground_truth`
- `height_in`, `width_in`, `length_in`, `weight_lbs`
- `dimensions_raw`, `weight_raw`, `is_edge_case`
